index.js

```diff
@@ -10,31 +10,41 @@ import { getImageHandler, imageGenerationHandler } from './routes/image';
 const { preflight, corsify } = createCors();
 
 // Create a new router
-const router = Router();
+const router = Router({ base: '/v1' });
 
-// CORS, see https://itty.dev/itty-router/cors
-router
-	// embed preflight upstream to handle all OPTIONS requests
-	.all('*', preflight);
-
-// chat completion
-router.post('/chat/completions', chatHandler);
-
-// legacy completions
-router.post('/completions', completionHandler);
-
-// embeddings
-router.post('/embeddings', embeddingsHandler);
-
-// audio transcriptions
-router.post('/audio/transcriptions', transcriptionHandler);
-
-// audio translations
-router.post('/audio/translations', translationHandler);
+function extractToken(authorizationHeader) {
+	if (authorizationHeader) {
+		const parts = authorizationHeader.split(' ');
+		if (parts.length === 2 && parts[0] === 'Bearer') {
+			return parts[1];
+		}
+	}
+	return null;
+}
+// MIDDLEWARE: withAuthenticatedUser - embeds user in Request or returns a 401
+const bearerAuthentication = (request, env) => {
```

Comment:
I think this is a great change, however I think we should have the option to leave the `ACCESS_TOKEN` blank and to then ignore this middleware if and only if the token is omitted from the config.